src/modalities/utils/mfu.py

```diff
@@ -18,124 +20,171 @@
+            otherwise None.
+        """
 
-    Returns:
-        (Number, optional): The accummulated theoretical peak performance of all GPUs,
-          or None if it cannot be calculated.
-    """
-    if torch.cuda.is_available() and torch.cuda.device_count() > 0:
-        precision = model.mixed_precision.param_dtype
-        if model.mixed_precision.reduce_dtype != precision or model.mixed_precision.buffer_dtype != precision:
-            warnings.warn(f"Could not get theoretical GPU peak performance for mixed precision type = {precision}.")
-            return None
+        if precision in [torch.float16, torch.bfloat16] and gpu_type in PEAK_PERFORMANCE:
+            return PEAK_PERFORMANCE[gpu_type]
         else:
-            device_name = torch.cuda.get_device_name()
-            if device_name.startswith("NVIDIA A100"):
-                single_gpu_peak_performance = _get_theoretical_gpu_peak_performance_single(precision, "A100")
-            elif device_name.startswith("NVIDIA H100"):
-                single_gpu_peak_performance = _get_theoretical_gpu_peak_performance_single(precision, "H100")
-            else:
-                warnings.warn(f"Could not get theoretical GPU peak performance for unknown device = {device_name}.")
-                return None
-            if single_gpu_peak_performance is None:
-                warnings.warn(
-                    f"Could not get theoretical GPU peak performance for device = {device_name} "
-                    f"and mixed precision type = {precision}."
-                )
+            return None
+
+    @staticmethod
+    def _get_theoretical_gpu_peak_performance(wrapped_model: FSDPX, world_size: int) -> Optional[Number]:
+        """
+        Calculates the accumulated theoretical peak performance based on all GPUs, i.e.,
+        #GPU=world_size, in units FLOPs / s for given gpu type.
+
+        Args:
+            model (FSDPX): The model for which to calculate the theoretical peak performance.
+            world_size (int): The number of GPUs used in parallel.
+
+        Returns:
+            (Number, optional): The accumulated theoretical peak performance of all GPUs,
+            or None if it cannot be calculated.
+        """
+        if isinstance(wrapped_model, FSDP1):
+            precision = wrapped_model.mixed_precision.param_dtype
+            if (
+                wrapped_model.mixed_precision.reduce_dtype != precision
+                or wrapped_model.mixed_precision.buffer_dtype != precision
+            ):
+                warnings.warn(f"Could not get theoretical GPU peak performance for mixed precision type = {precision}.")
                 return None
-            else:
-                return single_gpu_peak_performance * world_size
-    else:
-        return None
+        elif isinstance(wrapped_model, FSDP2):
+            warnings.warn("MFU is computed based on the assumption that bf16 precision is used.")
```

Comment:
Add a comment describing that dtype cannot be inferred from wrapped_model in FSDP2, and therefore, we assume that BF16 is used?